### PR TITLE
Deploy with docker

### PR DESCRIPTION
**Pull request by JackMeadDev**
* Has a default pipeline in place for learners to use/adjust

This does look to be more effort for learners to get working, but hopefully is a considerably more reliable release mechanism that's less prone to getting stuck, so definitely open to opinions on whether it's worthwhile but I think saving those few learners who aren't able to make progress at the cost of an extra few minutes setup for all is worthwhile

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Build image
+        run: docker build --tag ${DOCKERHUB_USERNAME}/m13-order-processing-app --target production .
+      - name: Docker login
+        run: echo $DOCKERHUB_TOKEN | docker login -u $DOCKERHUB_USERNAME --password-stdin
+      - name: Push image
+        run: docker push ${DOCKERHUB_USERNAME}/m13-order-processing-app
+      - name: Trigger deployment
+        run: curl -dHf -X POST "$AZURE_WEBHOOK"
```
Comment (Jonesey13) [inline .github/workflows/deploy.yml:29]: Do we need any of the `-dHf` flags here? (`-f` seems actively harmful)

File: app.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import requests
 app = Flask(__name__)
 app.config.from_object(Config)
+app.logger.setLevel(app.config['LOG_LEVEL'])
```
Comment (Jonesey13) [inline app.py:12]: Doesn't this conflict with the instruction to add 
```
import logging
logging.basicConfig(level=logging.INFO)
```
near the start of the exercise?

**Review by Jonesey13 — APPROVED** [review]
Looks good to me! Comments below are pretty minor.

**Comment by JackMeadDev** [comment]
Closing in favour of: https://github.com/corndeladmin/DevOps-Course-Workshop-Module-13-Learners/pull/1

**Pull request closed without merging**